src/groupByKey.ts

```diff
@@ -2,6 +2,41 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type Value = string | number;
+
+type Obj = {
+  [key: string]: Value;
+};
+
+export function groupByKey(items: Obj[], key: keyof Obj): GroupsMap<Obj> {
+  const result: GroupsMap<Obj> = {};
+  const moddedItems = items.map((item1) => {
+    const arrayForItems: Obj[] = [];
+
+    items.filter((item2) => {
+      if (item2[key] === item1[key]) {
+        arrayForItems.push(item2);
+
+        return true;
+      }
+
+      return false;
+    });
+
+    if (arrayForItems.length === 1) {
+      return arrayForItems[0];
+    }
```

Comment:
This condition seems unnecessary. If there is only one item in `arrayForItems`, it doesn't mean that we should return this item instead of the array. The function should always return an array of items grouped by the specified key, even if it contains only one item.